stan/math/rev/core/vari.hpp

```diff
@@ -528,6 +528,20 @@ class vari_view_eigen {
                                  derived().adj_.middleCols(start_col, n));
   }
 
+  /**
+   * Return an Array expression
+   */
+  inline auto array() const {
```

Comment:
Same as var, a basic test or two for: https://github.com/stan-dev/math/blob/develop/test/unit/math/rev/core/vari_test.cpp